libs/core/langchain_core/utils/function_calling.py

```diff
@@ -382,15 +384,20 @@ def convert_to_openai_tool(
             BaseTool. If a dictionary is passed in, it is assumed to already be a valid
             OpenAI tool, OpenAI function, or a JSON schema with top-level 'title' and
             'description' keys specified.
+        strict: If True, model output is guaranteed to exactly match the JSON Schema
+            provided in the function definition.
 
     Returns:
         A dict version of the passed in tool which is compatible with the
             OpenAI tool-calling API.
     """
     if isinstance(tool, dict) and tool.get("type") == "function" and "function" in tool:
         return tool
-    function = convert_to_openai_function(tool)
-    return {"type": "function", "function": function}
+    oai_function = convert_to_openai_function(tool)
+    oai_tool: Dict[str, Any] = {"type": "function", "function": oai_function}
+    if strict is not None:
+        oai_tool["strict"] = strict
+    return oai_tool
```

Comment:
At least in the JS SDK the `strict` field is inside the `function` field of a tool. Would double check in py

Reply:
great catch

Reply:
hm their api and py docs seem to suggest different things — maybe both are supported?
<img width="759" alt="Screenshot 2024-08-06 at 12 25 16 PM" src="https://github.com/user-attachments/assets/4a618364-086d-4cb5-8af7-929914102725">
<img width="878" alt="Screenshot 2024-08-06 at 12 25 25 PM" src="https://github.com/user-attachments/assets/118e6ab5-12d5-41fd-91bf-8d93683acecd">

Reply:
weird, I'm going to pass it via the `function` field in JS just to make types happy, and if they remove it'll cause issues we'll notice down the line.